src/sonic-ctrmgrd/ctrmgr/container_startup.py

```diff
@@ -114,9 +114,13 @@ def get_docker_id():
     # Read the container-id
     # Note: This script runs inside the context of container
     #
-    cmd = 'cat /proc/self/cgroup | grep -e ":memory:" | rev | cut -f1 -d\'/\' | rev'
-    proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
-    output = proc.communicate()[0].decode("utf-8")
+    cmd0 = ['cat', '/proc/self/cgroup']
+    cmd1 = ['grep', '-e', ":memory:"]
+    cmd2 = ['rev']
+    cmd3 = ['cut', '-f1', '-d', '/']
+    cmd4 = ['rev']
+    _, output = getstatusoutput_noshell_pipe(cmd0, cmd1, cmd2, cmd3, cmd4)
+    # output = proc.communicate()[0].decode("utf-8")
```

Comment:
># output = proc.communicate()[0].decode("utf-8") [](http://example.com/codeflow?start=4&length=48)

Do you need this line of code? if not, delete it.

Reply:
@lixiaoyuner Can you please review ?